### ISSUE TITLE
push() and pop() don't seem to apply to strokeWeight() in webgl

### ISSUE THREAD

**Issue by JulesGalbraith** [processing/p5.js#3294]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [x ] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: v0.7.1 August 10, 2018 
Chrome: 70.0.3538.77 (Build officiel) (64 bits) (cohort: Stable)
Javascript: V8 7.0.276.32
OS: Windows 10

This is a beginner code that creates a simulacrum of a rotating sun/earth/moon. I tried to make the stroke weight of the sun thicker than that of the sun/moon. the visual format/ style for each ellipsoid are within push/pop. however, while the colour of each remained discrete, the thicker stroke of the sun applied to each ellipsoid
```js
var angleX = 0.0;
var angleY = 0.0;
var angleZ = 0.0;
var numStars = 1000;

function setup () {
  createCanvas(800,600,WEBGL);
  }


function draw() {
  background(0);

  for (var i = 0; i < numStars; i++) {
      var x = random(-width,width);
      var y = random(-height,height);
      var starSize = random(1,2);
      stroke(255);
      rect(x,y,starSize,starSize);
    }

  rotateY(angleY);
//sun
  push();
  fill(237, 4, 39);
  //here is the bug!
  strokeWeight(5);
  stroke(249, 208, 57);
  ellipsoid(100,100,100);
  pop();
  angleY += 0.03;
  translate(150,0,150);
  rotateX(angleX);
//earth
  push();
  fill(9, 51, 193);
  stroke(94, 128, 242);
  ellipsoid(60,60,60);
  pop();
  angleX += 0.03
  translate(50,50,50);
  rotateZ(angleZ);
//moon
  push();
  fill(255);
  stroke(150);
  ellipsoid(20,20,20);
  pop();
  angleZ += 0.03;

}
```

**Comment by Spongman** [comment]
this is another bug due to the fact that webgl renderer state is kept in the shaders, and not in the renderers. this would have been fixed by #2670.

**Comment by stalgiag** [comment]
Thanks for filing the issues. This wasw fixed with #3535.
